Essentials/src/com/earth2me/essentials/commands/Commandsethome.java

```diff
@@ -41,19 +43,41 @@ public void run(final Server server, final User user, final String commandLabel,
                 }
             }
         }
+
         if (checkHomeLimit(user, usersHome, name)) {
             name = "home";
         }
         if ("bed".equals(name) || NumberUtil.isInt(name)) {
             throw new NoSuchFieldException(tl("invalidHomeName"));
         }
+        if (ess.getSettings().isSetSameHomeByConfirm()) {
+            UUID uuid = user.getBase().getUniqueId();
+            for (String h : usersHome.getHomes()) {
+                if (h.equalsIgnoreCase(name)) {
+                    if (!confirmSetHome.contains(uuid)) {
```

Comment:
Assuming you change `confirmSetHome` to be a Set, 57 and 58 can be changed to just:

```
if (confirmSetHome.add(uuid)) {
```